department-of-veterans-affairs/vets-website
# 88477 Welcome component: use preferred name from user profile

## Are you removing, renaming or moving a folder in this PR?


### :warning: TeamSites :warning:
Examples of a TeamSite: https://va.gov/health and https://benefits.va.gov/benefits/. This scenario is also referred to as the "injected" header and footer. You can reach out in the `#sitewide-public-websites` Slack channel for questions.

Did you change site-wide styles, platform utilities or other infrastructure?

## Summary
Use the Preferred name attribute that was recently added to the `/v0/user` API (sourced from the MPI `preferred_name`).

This PR:
- adds preferred name to the state
- updates the Welcome component to use preferredName from the `/v0/user` API
- updates Profile link to use `<a>` element  instead of `<va-link />` component and add link click tracking by:
  - Google Analytics navigation with 'nav-link-click' event,
  - Datadog tracking of `data-dd-action-name`

## Related issue(s)

- https://github.com/department-of-veterans-affairs/va.gov-team/issues/88477

## Testing done
- updated unit tests for GA and Datadog tracking

- Local testing:
  - Turn on `mhvLandingPagePersonalization` feature flag on `mhv-landing-page/mock/api/feature-toggles/index.js`
  - Run vets-website locally `yarn watch --env entry=mhv-landing-page`
  - Run mock apis on another terminal `yarn mock-api --responses src/applications/mhv-landing-page/mocks/api/index.js`
  - Navigate to MHV Landing Page and verify that preferred name is shown correctly
  - Inspect network activity using Chrome devtools to verify that PreferredName attribute is returned on the `user` API and that `personal_information` API is not fetched 


## Screenshots

Screenshots are expected to remain the same:

|         | Before | After |
| ------- | ------ | ----- |
| Mobile  |   <img width="301" alt="Screenshot 2024-11-22 at 10 55 01 AM" src="https://github.com/user-attachments/assets/2b210ee1-3880-4b96-8176-9ae5381ea33e">     |    <img width="301" alt="Screenshot 2024-11-22 at 10 06 21 AM" src="https://github.com/user-attachments/assets/e3a836cb-3265-4df9-8eec-782c202afeb4">   |
| Desktop |    <img width="652" alt="Screenshot 2024-11-22 at 10 55 39 AM" src="https://github.com/user-attachments/assets/cc6b1145-a2bb-46c3-9681-b0b5c5ad9b43"> | <img width="657" alt="Screenshot 2024-11-22 at 10 05 37 AM" src="https://github.com/user-attachments/assets/52efa8af-bb3a-4e0e-aea5-3e96cd5d6117">      |

## What areas of the site does it impact?
- Adds perferredName to the user state
- MHV landing page


## Acceptance criteria

### Quality Assurance & Testing


### Error Handling


### Authentication


## Requested Feedback
